src/cmdstan/command.hpp

```diff
@@ -125,12 +125,16 @@ int command(int argc, const char *argv[]) {
 #endif
   argument_parser parser(valid_arguments);
   int err_code = parser.parse_args(argc, argv, info, err);
-  if (err_code != 0) {
-    std::cout << "Failed to parse arguments, terminating Stan" << std::endl;
+  if (err_code == stan::services::error_codes::CONFIG) {
+    std::cerr << "Failed to parse command arguments, cannot run model."
+              << std::endl;
+    return err_code;
+  } else if (err_code != 0) {
+    std::cerr << "Unexpected failure, quitting." << std::endl;
     return err_code;
   }
   if (parser.help_printed())
-    return err_code;
+    return stan::services::error_codes::OK;
```

Comment:
It looks like `rm` sets an error code here, so maybe we should too?

```
$ rm
rm: missing operand
Try 'rm --help' for more information.
$ echo $?
1
```

Reply:
right, but this is more like the following:

```
~> make
make: *** No targets specified and no makefile found.  Stop.
~> echo $?
2
~> make -h
Usage: make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
 ....
~> echo $?
0
~> 
```

Reply:
lines 136-137 are correct.  changed line 128 - return code `USAGE` set when no args are given - this will result in help message being printed; and then program will exit.  so if you get to line 136, user requested help to be printed, help was printed, all good.  cf above example